packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts

```diff
@@ -17,4 +41,112 @@ test.describe('PieCookieBanner - Component tests', () => {
+            },
+        });
+
+        const cookieBanner = await page.locator(componentSelector);
+
+        // Act
+        await page.locator(acceptAllSelector).click();
+
+        // Assert
+        expect(events).toHaveLength(1);
+        expect(cookieBanner).not.toBeVisible();
+    });
+
+    test('clicking "Necessary only" emits the correct event and closes the cookie banner', async ({ mount, page }) => {
+        // Arrange
+        const events : Array<Event> = [];
+
+        await mount(PieCookieBanner, {
+            props: {} as CookieBannerProps,
+
+            on: {
+                [ON_COOKIE_BANNER_NECESSARY_ONLY]: (event: Event) => events.push(event),
+            },
+        });
+
+        const cookieBanner = await page.locator(componentSelector);
+
+        // Act
+        await page.locator(necessaryOnlySelector).click();
+
+        // Assert
+        expect(events).toHaveLength(1);
+        expect(cookieBanner).not.toBeVisible();
+    });
+
+    test('clicking "Manage preferences" emits the correct event, opens the modal and hides the cookie banner', async ({ mount, page }) => {
+        // Arrange
+        const events : Array<Event> = [];
+
+        await mount(PieCookieBanner, {
+            props: {} as CookieBannerProps,
+
+            on: {
+                [ON_COOKIE_BANNER_MANAGE_PREFS]: (event: Event) => events.push(event),
+            },
+        });
+
+        const cookieBanner = await page.locator(componentSelector);
+
+        // Act
+        await page.locator(managePreferencesSelector).click();
+
+        // Assert
+        const modal = await page.locator(modalSelector);
+
+        expect(modal).toBeVisible();
+        expect(cookieBanner).not.toBeVisible();
+        expect(events).toHaveLength(1);
+    });
+
+    test('clicking the back button in "Manage preferences" closes the modal and re-displays the cookie banner', async ({ mount, page }) => {
+        // Arrange
+        await mount(PieCookieBanner, {
+            props: {} as CookieBannerProps,
+        });
+
+        const cookieBanner = await page.locator(componentSelector);
+
+        // Act
+        await page.locator(managePreferencesSelector).click();
+        await page.locator(modalBackButtonSelector).click();
+
+        // Assert
+        const modal = await page.locator(modalSelector);
+
+        expect(modal).not.toBeVisible();
+        expect(cookieBanner).toBeVisible();
+    });
+
+    // TODO: Once we add preference saving logic we should update this test to ensure it behaves as expected
+    test('clicking save in "Manage preferences" closes the modal and closes the cookie banner', async ({ mount, page }) => {
+        // Arrange
+        await mount(PieCookieBanner, {
+            props: {} as CookieBannerProps,
+        });
+
+        const cookieBanner = await page.locator(componentSelector);
+
+        // Act
+        await page.locator(managePreferencesSelector).click();
```

Comment:
Not requesting changes, but I think you can do `await page.click(selector)` 😄

Reply:
perfect! I'll update in the follow up PR if ok :)